### PR TITLE
Added Library tab for web debug

### PR DESCRIPTION
**Pull request by cdavidjean**
- Added async updating of library tab
- Added hack to be able to open packages file for package not in analyzer root.

### CODE REVIEW

File: lib/debug/debugger.dart [hunk 1]
```diff
@@ -160,8 +163,26 @@ class UriResolver implements Disposable {
     if (_uriToPath.containsKey(uri)) return _uriToPath[uri];
 
     String contextId = await _completer.future;
-    MapUriResult result = await analysisServer.server.execution.mapUri(contextId, uri: uri);
-    String path = result.file;
+    String path;
+    try {
```
Comment (devoncarew) [inline lib/debug/debugger.dart]: Hmm, I'm curious what this is doing. You're using the analysis server to resolve a uri to a file path? There is an API for that, in the debugger domain.

I may not understand what this is trying to do though - I'd suggest adding some descriptive comments :)

Comment (devoncarew) [reply lib/debug/debugger.dart]: It is at least very creative :)

File: lib/debug/debugger_ui.dart [hunk 2]
```diff
@@ -120,6 +121,9 @@ class DebuggerView extends DockedView {
     subs.add(connection.onPaused.listen(_handleIsolatePaused));
     subs.add(connection.onResumed.listen(_handleIsolateResumed));
     subs.add(connection.isolates.onRemoved.listen(_handleIsolateTerminated));
+
+    // Optional events
+    subs.add(connection.onLibrariesChanged?.listen(_handleLibrariesChanged));
```
Comment (devoncarew) [inline lib/debug/debugger_ui.dart]: Ah, this will add a `null` subscription sometimes. You may instead want to re-write to use an if statement.

Comment (cdavidjean) [reply lib/debug/debugger_ui.dart]: Done.

File: lib/debug/debugger.dart (hunk 1 above)
Comment (cdavidjean) [reply lib/debug/debugger.dart]: :)  Added comment.  If you think it is an analyzer bug, please let me know.

**Pull request merged**